tests/App/Datagrid/Pager.php

```diff
@@ -41,40 +42,80 @@ public function setMaxPerPage($max): void
     {
     }
 
-    public function setPage($page): void
+    public function getPage(): int
     {
+        return 1;
     }
 
-    public function setQuery($query): void
+    public function setPage($page): void
     {
     }
 
-    public function getResults(): array
+    public function getNextPage(): int
     {
-        return $this->repository->all();
+        return 1;
     }
 
-    public function setMaxPageLinks($maxPageLinks): void
+    public function getPreviousPage(): int
     {
+        return 1;
     }
 
-    public function getMaxPageLinks()
+    public function getFirstPage(): int
     {
         return 1;
     }
 
-    public function getPage()
+    public function isFirstPage(): bool
     {
-        return 1;
+        return true;
     }
 
-    public function isLastPage()
+    public function getLastPage(): int
+    {
+        return 2;
+    }
+
+    public function isLastPage(): bool
     {
         return true;
     }
 
+    public function getQuery($query): ProxyQueryInterface
+    {
+        return new ProxyQuery();
+    }
+
+    public function setQuery($query): void
+    {
+    }
+
+    public function haveToPaginate(): bool
+    {
+        return false;
+    }
+
+    public function getResults(): array
+    {
+        return $this->repository->all();
+    }
+
     public function getNbResults()
```

Comment:
Maybe not related to this PR, but what is a **Nb**Result? We should rename this method

Reply:
Nb for Number (or Number of)

There is a similar discussion here: https://github.com/sonata-project/SonataAdminBundle/pull/6541#discussion_r532515190